### PR TITLE
Add map frame rate and current frame numbers to QgsMapSettings/QgsRenderContext

### PR DESCRIPTION
**Pull request by nyalldawson**
Makes this information available for symbol rendering logic when a map is being rendered as part of an animation

Useful for either temporal controller set to animation mode, or 3rd party animation creation plugins

### CODE REVIEW

**Comment by nyalldawson** [comment]
Ping @timlinux

File: src/gui/qgsmapcanvas.cpp [hunk 1]
```diff
@@ -644,6 +675,17 @@ void QgsMapCanvas::refreshMap()
   stopPreviewJobs();
 
   mSettings.setExpressionContext( createExpressionContext() );
+
+  // if using the temporal controller in animation mode, get the frame settings from that
+  if ( QgsTemporalNavigationObject *to = dynamic_cast < QgsTemporalNavigationObject * >( mController ) )
```
Comment (nirvn) [inline src/gui/qgsmapcanvas.cpp:680]: @nyalldawson , you've called previous pointers "no", while this one is "to". Could you harmonize and maybe use a slightly expanded name, like navObj or something? :)

**Review by nirvn — APPROVED** [review]
Looking good, just one comment that'd be nice to have addressed.

File: src/gui/qgsmapcanvas.cpp (hunk 1 above)
Comment (nyalldawson) [reply src/gui/qgsmapcanvas.cpp:680]: Done

**Pull request merged**